## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 463627`
Commit: [`b0282baed996e1d07dc5ea315a5606f800d67ecd`](https://github.com/NixOS/nixpkgs/commit/b0282baed996e1d07dc5ea315a5606f800d67ecd) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/b0282baed996e1d07dc5ea315a5606f800d67ecd..pull/463627/head))
Merge: [`16dcede61ef00f4ad2d0d05096b252f5d4dada4b`](https://github.com/NixOS/nixpkgs/commit/16dcede61ef00f4ad2d0d05096b252f5d4dada4b)

Logs: https://github.com/yzhou216/nixpkgs-review-gha/actions/runs/20611083814

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-463627-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/s2d19anvca3b51qbckmhxsfky0cmvmc2-c-ascii-render-0-unstable-2025-11-18
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-463627-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/vbsavwjd4dv6vzxk24507kz087j5yg6p-c-ascii-render-0-unstable-2025-11-18
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>c-ascii-render</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>c-ascii-render</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>c-ascii-render</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>c-ascii-render</summary>
<pre>@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/ddxxigbjkpya7bmd403g2djsrfsrx8g9-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/a5kqrf4vqb287bfs1y2k9pivy0qj729g-bash-5.3p3/bin/bash
mkdir -p build/obj
cc -Wall -Wextra -Werror -std=c11 -pedantic -I./include -g -O0 -coverage -c src/audio.c -o build/obj/audio.o
cc -Wall -Wextra -Werror -std=c11 -pedantic -I./include -g -O0 -coverage -c src/input.c -o build/obj/input.o
cc -Wall -Wextra -Werror -std=c11 -pedantic -I./include -g -O0 -coverage -c src/main.c -o build/obj/main.o
src/main.c:140:12: error: use of undeclared identifier &#x27;SIGWINCH&#x27;
  140 |     signal(SIGWINCH, sigwinch_handler);
      |            ^~~~~~~~
1 error generated.
make: *** [Makefile:43: build/obj/main.o] Error 1</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>c-ascii-render</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>c-ascii-render</summary>
<pre>source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
no configure script, doing nothing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: SHELL=/nix/store/yi3x6xpilh0bq0bgb61imy6gpv8b1la4-bash-5.3p3/bin/bash
mkdir -p build/obj
cc -Wall -Wextra -Werror -std=c11 -pedantic -I./include -g -O0 -coverage -c src/audio.c -o build/obj/audio.o
cc -Wall -Wextra -Werror -std=c11 -pedantic -I./include -g -O0 -coverage -c src/input.c -o build/obj/input.o
cc -Wall -Wextra -Werror -std=c11 -pedantic -I./include -g -O0 -coverage -c src/main.c -o build/obj/main.o
src/main.c:140:12: error: use of undeclared identifier &#x27;SIGWINCH&#x27;
  140 |     signal(SIGWINCH, sigwinch_handler);
      |            ^~~~~~~~
1 error generated.
make: *** [Makefile:43: build/obj/main.o] Error 1</pre>
</details>
</details>

--- a/pkgs/by-name/c-/c-ascii-render/package.nix
+++ b/pkgs/by-name/c-/c-ascii-render/package.nix
@@ -1,0 +1,40 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation {
+  pname = "c-ascii-render";
+  version = "0-unstable-2025-11-18";
+
+  src = fetchFromGitHub {
+    owner = "Lallapallooza";
+    repo = "c_ascii_render";
+    rev = "c1894b50488c6ba75e33734d6b3d4a3397ac1fb5";
+    hash = "sha256-sqGKUbv8OCWAs7Rxe2H6+xHQJivBkGM7K13GdNuPVd8=";
+  };
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace-fail "CC = gcc" "CC = ${stdenv.cc.targetPrefix}cc" \
+      --replace-fail '/usr/local' '${placeholder "out"}'
+  '';
+
+  env.NIX_CFLAGS_COMPILE =
+    "-Wno-error"
+    # otherwise does not find SIGWINCH
+    + lib.optionalString stdenv.hostPlatform.isDarwin " -D_DARWIN_C_SOURCE";
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Simple ASCII Render using Pure C";
+    homepage = "https://github.com/Lallapallooza/c_ascii_render";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ yiyu ];
+    mainProgram = "c-ascii-render";
+    platforms = lib.platforms.all;
+  };
+}

